src/components/Missions.jsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect } from 'react';
+import Container from 'react-bootstrap/Container';
+import Table from 'react-bootstrap/Table';
+import Badge from 'react-bootstrap/Badge';
+import Button from 'react-bootstrap/Button';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchMissions, joinMission, leaveMission } from '../redux/missions/missions';
+
+const Missions = () => {
+  const dispatch = useDispatch();
+  const missions = useSelector((state) => state.missions);
+
+  useEffect(() => {
+    if (!missions.length) {
+      dispatch(fetchMissions);
+    }
+  }, []);
+
+  const handleJoin = (id) => dispatch(joinMission(id));
+  const handleLeave = (id) => dispatch(leaveMission(id));
+
+  return (
+    <Container style={{ overflowX: 'auto' }}>
+      <Table striped bordered hover>
+        <thead>
+          <tr>
+            <th>Mission</th>
+            <th>Description</th>
+            <th>Status</th>
+            <th>{'  '}</th>
+          </tr>
+        </thead>
+        <tbody>
+          {missions.map(({
+            id, name, description, reserved,
+          }) => (
+            <tr key={id}>
+              <td><p className="fw-bold">{name}</p></td>
+              <td><p>{description}</p></td>
+              <td className="align-middle">
+                {reserved && <Badge bg="info">Active Member</Badge>}
+                {!reserved && <Badge bg="secondary">NOT A MEMBER</Badge>}
+              </td>
+              <td className="col-2 align-middle text-center">
+                {reserved
+                  && (
+                  <Button
+                    size="sm"
+                    variant="outline-danger"
+                    onClick={() => handleLeave(id)}
+                  >
+                    Leave Mission
+                  </Button>
+                  )}
+                {!reserved
+                  && (
+                  <Button
+                    size="sm"
+                    variant="outline-secondary"
+                    onClick={() => handleJoin(id)}
+                  >
+                    Join Mission
+                  </Button>
+                  )}
+              </td>
+            </tr>
+          ))}
+        </tbody>
```

Comment:
- [ ] When fetching data from an API, the response might get delayed or it might return an error, kindly update this component to conditionally render `Missions` and you may as well display a message if an error occurred.

Reply:
Thank you for the feedback. It is now done as suggested.